channels/pkg/channels/channel_version.go

```diff
@@ -39,6 +40,17 @@ type ChannelVersion struct {
 	Channel *string `json:"channel,omitempty"`
 }
 
+func stringValue(s *string) string {
+	if s == nil {
+		return ""
+	}
+	return *s
+}
+
+func (c *ChannelVersion) String() {
```

Comment:
It allows for a difference between not-set and set-to-empty.  It avoids the need for per-field logic to know whether empty means empty or not-set.